kura/org.eclipse.kura.ble.provider/src/main/java/org/eclipse/kura/internal/ble/util/BluetoothProcess.java

```diff
@@ -162,11 +179,12 @@ private void readBTSnoopStreamFully(InputStream is, BTSnoopListener listener) th
         while (this.btSnoopReady) {
             if (is != null) {
                 byte[] packet = this.parser.readRecord();
-                listener.processBTSnoopRecord(packet);
+                if (packet.length > 0) {
+                    listener.processBTSnoopRecord(packet);
+                }
             }
         }
-        closeStreams();
-        logger.debug("End of stream!");
```

Comment:
Why did you remove the streams close?

Reply:
The while() loop here doesn't quit unless you call destroyBTSnoop(). This function includes a call to closeStreams() already so calling it again here is unnecessary from what I can see.